### PR TITLE
Added Markdown preview in the box

### PR DESCRIPTION
**Pull request by john-ghatas**
https://alice.its.cern.ch/jira/secure/RapidBoard.jspa?rapidView=344&projectKey=O2B&view=detail&selectedIssue=O2B-248
Note: Adding files is not yet supported because of the same-origin policy

### CODE REVIEW

File: lib/public/index.html [hunk 1]
```diff
@@ -11,10 +11,64 @@
...
     <link rel="stylesheet" href="/app.css">
-
+    
     <script src="/js/src/compatibility.js"></script>
 </head>
 
+  <!-- ----------------------------------------------------------------------------------- -->
+  <!-- CodeMirror -->
+
+  <link rel="stylesheet" href="./assets/SmartEditor/codemirror/lib/codemirror.css">
```
Comment (mvegter) [inline lib/public/index.html:21]: Do not use relative paths here

File: lib/public/index.html [hunk 2]
```diff
@@ -34,11 +88,11 @@
...
-        const model = new Model();
+        const model = new Model(HyperMD);
         const debug = true; // shows when redraw is done
         mount(document.body, view, model, debug);
 
         // Expose model to interact with it the browser's console
         window.model = model;
     </script>
-</body>
+</body>
```
Comment (mvegter) [inline lib/public/index.html:98]: restore the empty line

File: lib/public/index.html [hunk 3]
```diff
@@ -11,10 +11,64 @@
     <link rel="shortcut icon" href="/favicon.ico" />
     <link rel="stylesheet" href="/css/src/bootstrap.css">
     <link rel="stylesheet" href="/app.css">
-
+
```
Comment (mvegter) [inline lib/public/index.html:14]: cleanup here too, please

File: lib/public/views/Logs/Create/index.js [hunk 4]
```diff
@@ -25,6 +25,13 @@ const createScreen = (model) => {
     const errorMessages = model.logs.getErrorMessages()
         || [{ detail: 'There was a problem with posting the log, try again later' }];
 
+    const preview = h('', [
```
Comment (mvegter) [inline lib/public/views/Logs/Create/index.js:28]: Shouldn't this be an arrow function instead of a variable?

File: lib/public/Model.js [hunk 5]
```diff
@@ -58,6 +61,8 @@ export default class Model extends Observable {
      * @returns {vnode} The page to be loaded
      */
     handleLocationChange() {
+        // Clear the editor before navigating
+        this.logs.clearEditor();
```
Comment (mvegter) [inline lib/public/Model.js:65]: Shouldn't this be in a onremove method?

File: lib/public/index.html (hunk 1 above)
Comment (john-ghatas) [reply lib/public/index.html:21]: Any specific reason for that?

File: lib/public/views/Logs/Create/index.js (hunk 4 above)
Comment (john-ghatas) [reply lib/public/views/Logs/Create/index.js:28]: Done

File: lib/public/index.html (hunk 2 above)
Comment (john-ghatas) [reply lib/public/index.html:98]: Done

File: lib/public/Model.js (hunk 5 above)
Comment (john-ghatas) [reply lib/public/Model.js:65]: Calling the onremove like this: 
image
Doesn't clear the editor (it's never called upon

**Comment by john-ghatas** [comment]
It's a mix with minified js files or full fledged js files (depending on the plug), couldn't find any minified versions so far.

File: lib/public/index.html (hunk 1 above)
Comment (mvegter) [reply lib/public/index.html]: As using relative paths can break while absolute paths wont.

File: lib/public/Model.js (hunk 5 above)
Comment (mvegter) [reply lib/public/Model.js:65]: Then it is most likely on the wrong component as the onremove is called in the tag overview/detail and in logs (see the open PR).

**Comment by mvegter** [comment]
I found minified version by changing the .js extension to .min.js of the imports in https://github.com/laobubu/HyperMD/blob/master/docs/examples/ai1.html

File: lib/public/Model.js (hunk 5 above)
Comment (john-ghatas) [reply lib/public/Model.js]: Fixed :)

File: lib/public/index.html (hunk 1 above)
Comment (john-ghatas) [reply lib/public/index.html]: Done as well

**Comment by john-ghatas** [comment]
Added the minified versions as much as I could, see [the latest commit](https://github.com/AliceO2Group/Bookkeeping/pull/95/commits/41dfb5db086ca30006518d4254a2472122bede81)

File: .nycrc [hunk 6]
```diff
@@ -7,6 +7,7 @@
         "lib/main.js",
         "lib/database/migrations/*.js",
         "lib/database/seeders/*.js",
-        "test/**/*.js"
+        "test/**/*.js",
+        "lib/public/assets/**"
```
Comment (mvegter) [inline .nycrc:11]: Sort these in alphabetical order

Comment (john-ghatas) [reply .nycrc]: Done

File: lib/public/Model.js (hunk 5 above)
Comment (john-ghatas) [reply lib/public/Model.js]: @mvegter

**Pull request merged**